docs: Redwood release notes first cut. (#507)
openedx/docs.openedx.org: source/community/release_notes/redwood.rst

* ``BADGR_ENABLE_NOTIFICATIONS``, ``SEND_CERTIFICATE_REVOKED_SIGNAL``,
* ``blockstore.use_blockstore_app_api``,
* ``contentstore.enable_copy_paste_units``,
* ``course_apps.proctoring_settings_modal_view``,
* ``course_live.enable_big_blue_button``,
* ``course_live.enable_course_live``,
* ``courseware.learning_assistant``,
* ``discussions.enable_learners_stats``,
* ``discussions.enable_learners_tab_in_discussions_mfe``,
* ``discussions.enable_moderation_reason_codes``,
* ``discussions.enable_reported_content_email_notifications``,
* ``learner_recommendations.enable_course_about_page_recommendations``,
* ``learner_recommendations.enable_dashboard_recommendations``,
* ``student.enable_2u_recommendations``,
* ``student.enable_amplitude_recommendations``,
* ``student.enable_fallback_recommendations``,
* ``blockstore.use_blockstore_app_api`` are also not configurable anymore.

Developer Experience
********************

-  *Asset Processing (webpack, collectstatic, etc.) using Paver Commands in edx-platform is now Deprecated and will not be available in Sumac*
   -  *Non deployment paver commands will be removed by Sumac*

Researcher & Data Experiences
*****************************

> 

Reviewer: ```suggestion
`Aspects <https://docs.openedx.org/projects/openedx-aspects/en/latest/index.html>`_ 
is an analytics system for the Open edX platform, bringing actionable data
about course and learner performance to instructors and site operators. It is primarily
a Tutor plugin that ties together data from the Open edX learning management system
and Studio using open source tools to aggregate and transform learning traces into data
visualizations.

See the `Aspects configuration documentation <https://docs.openedx.org/projects/openedx-aspects/en/latest/how-tos/production_configuration.html>`_
to learn about setting up Aspects for your production environment.
```

* `EVENT_BUS_PRODUCER_CONFIG['org.openedx.content_authoring.course.catalog_info.changed.v1'] <https://github.com/openedx/edx-platform/blob/7d11c889bbbf55dfa69c734122de72d83c1893bf/cms/envs/common.py#L2849>`_
* `EVENT_BUS_PRODUCER_CONFIG['org.openedx.learning.course.unenrollment.completed.v1'] <https://github.com/openedx/edx-platform/blob/7d11c889bbbf55dfa69c734122de72d83c1893bf/lms/envs/common.py#L5428>`_
* `EVENT_BUS_PRODUCER_CONFIG['org.openedx.learning.xblock.skill.verified.v1'] <https://github.com/openedx/edx-platform/blob/7d11c889bbbf55dfa69c734122de72d83c1893bf/lms/envs/common.py#L5443>`_

Other Operator Changes
======================

-  The default minimum password length has been updated from 2
   characters to 8 characters. `(PR) <https://github.com/openedx/edx-platform/pull/33373>`_.

   -  For users with an existing password, this change alone will not
      force them to update it. However if they reset their password or go
      to change it, they'll have to conform to the new guidelines. If you
      would like to force people to update their password, you

>      should take a look at `the password_policy plugin and its settings <https://github.com/openedx/edx-platform/blob/2033dcf6ace133719aaeb72dc5dd6ee521a7ac42/openedx/core/djangoapps/password_policy/settings/common.py#L13>`_

Reviewer: ```suggestion
      should take a look at `the password_policy plugin and its settings <https://github.com/openedx/edx-platform/blob/2033dcf6ace133719aaeb72dc5dd6ee521a7ac42/openedx/core/djangoapps/password_policy/settings/common.py#L13>`_
```

-  The Credentials service `updated some requirements <https://github.com/openedx/credentials/commit/1cd7c25c04a955aa9aaa263fb40ebd3f73d0937e>`_ and may have implications for anyone
   who has a massive ``usersocialauth`` table.  This is because that
   table grows endlessly, and the migrations on the table caused by
   updating the ``social-auth-app-django`` package can run out of
   memory. If maintainers have migration failures on this upgrade, they
   should run the management command `truncate_social_auth <https://github.com/openedx/credentials/blob/master/credentials/apps/core/management/commands/truncate_social_auth.py>`_.

   -  This will remove all entries from the ``usersocialauth`` table
      that haven't been updated in 90 days, which makes the size of the
      table tractable for the dependency's migration. This is harmless
      in the ``Credentials`` IDA.

-  The scripts related to user retirement across all services
   have been moved to the ``edx-platform`` repository. If you’ve been
   using the `unsupported tubular repository <https://github.com/openedx-unsupported/tubular>`_ to run retirement scripts you should update
   your code.

>    -  *Relevant Tickets*

Reviewer: ```suggestion
   -  Relevant Tickets
```

>    -  *Relevant Tickets*

Reviewer: ```suggestion
   -  Relevant Tickets

```

      libraries in Studio; it will display regular progress indicators
      while it is running. We are interested in hearing how long it
      takes for you - please share your experience (see next bullet).
      This command reads from MySQL/MongoDB but does not write to them;
      it only writes to Meilisearch. Once the indexing has completed, it
      should not be necessary to run it again; from that point forward,
      the indexes will be updated automatically as needed.

   -  Please share your feedback about Meilisearch, indexing, and
      operations in `this Discourse thread <https://discuss.openedx.org/t/is-meilisearch-a-viable-upgrade-alternative-to-opensearch/12400>`_ or the `#ops <https://openedx.slack.com/archives/C08B4LZEZ>`_ Slack channel. Please share feedback about
      the new course search feature in general `in the discussion forums <https://discuss.openedx.org/t/feedback-thread-new-course-search/13076>`_ or in the `#wg-product-core <https://openedx.slack.com/archives/C057J2D1WU9>`_ Slack channel.

Deprecations & Removals
***********************

-  Badges app has been deprecated and removed from ``edx-platform``.

>    See `[DEPR]: lms/djangoapps/badges <https://github.com/openedx/edx-platform/issues/31541>`_ .*

Reviewer: ```suggestion
   See `[DEPR]: lms/djangoapps/badges <https://github.com/openedx/edx-platform/issues/31541>`_ .
```

> -  - In edxapp, the Waffle switch ``ip.legacy`` is removed. See `[DEPR]: legacy_ip code and Waffle switch <https://github.com/openedx/edx-platform/issues/33733>`_ .*

Reviewer: ```suggestion
-  - In edxapp, the Waffle switch ``ip.legacy`` is removed. See `[DEPR]: legacy_ip code and Waffle switch <https://github.com/openedx/edx-platform/issues/33733>`_ .
```

>    -  *Any deployment that has been relying on this legacy IP address

Reviewer: ```suggestion
   -  Any deployment that has been relying on this legacy IP address
```

      option will need to switch to setting
      ``CLOSEST_CLIENT_IP_FROM_HEADERS`` appropriately. See `Nutmeg Announcement <https://openedx.atlassian.net/wiki/spaces/COMM/pages/3205201949/Nutmeg#CLOSEST_CLIENT_IP_FROM_HEADERS>`_ for

>       details.*

Reviewer: ```suggestion
      details.
```

-  Asset Processing (webpack, collectstatic, etc.) using Paver Commands in edx-platform is now Deprecated and will not be available in Sumac

>    - `[DEPR]: Asset processing in Paver <https://github.com/openedx/edx-platform/issues/31895>`_.

Reviewer: ```suggestion

   - `[DEPR]: Asset processing in Paver`_.
```

>    -  *Non deployment paver commands will be removed by Sumac*

Reviewer: ```suggestion
   -  *Non deployment paver commands will be removed by Sumac*

```

> -  *The django-splash capability was removed from edx-platform and the relevant code has been archived.*

Reviewer: ```suggestion
-  The ``django-splash`` capability was removed from ``edx-platform`` and the relevant code has been archived.

```

   - https://github.com/openedx/public-engineering/issues/224

> 

Reviewer: Deprecated 
```suggestion
Flags and toggles removed in Redwood
==================================== 

```

* ``BADGING_BACKEND``, ``BADGR_BASE_URL``, ``BADGR_ISSUER_SLUG``,
* ``BADGR_USERNAME``, ``BADGR_PASSWORD``,
* ``BADGR_TOKENS_CACHE_KEY``, ``BADGR_TIMEOUT``,
* ``BADGR_ENABLE_NOTIFICATIONS``, ``SEND_CERTIFICATE_REVOKED_SIGNAL``,
* ``blockstore.use_blockstore_app_api``,
* ``contentstore.enable_copy_paste_units``,
* ``course_apps.proctoring_settings_modal_view``,
* ``course_live.enable_big_blue_button``,
* ``course_live.enable_course_live``,
* ``courseware.learning_assistant``,
* ``discussions.enable_learners_stats``,
* ``discussions.enable_learners_tab_in_discussions_mfe``,
* ``discussions.enable_moderation_reason_codes``,
* ``discussions.enable_reported_content_email_notifications``,
* ``learner_recommendations.enable_course_about_page_recommendations``,
* ``learner_recommendations.enable_dashboard_recommendations``,
* ``student.enable_2u_recommendations``,
* ``student.enable_amplitude_recommendations``,
* ``student.enable_fallback_recommendations``,
* ``blockstore.use_blockstore_app_api`` are also not configurable anymore.

Developer Experience
********************

> -  *Asset Processing (webpack, collectstatic, etc.) using Paver Commands in edx-platform is now Deprecated and will not be available in Sumac*

Reviewer: ```suggestion
-  Asset Processing (webpack, collectstatic, etc.) using Paver Commands in edx-platform is now Deprecated and will not be available in Sumac

```

>    -  *Non deployment paver commands will be removed by Sumac*

Reviewer: ```suggestion
   -    - `[DEPR]: Asset processing in Paver`_.
   -  Non deployment paver commands will be removed by Sumac

.. _ [DEPR]: Asset processing in Paver: https://github.com/openedx/edx-platform/issues/31895
```

Researcher & Data Experiences
*****************************

`Aspects <https://docs.openedx.org/projects/openedx-aspects/en/latest/index.html>`_ 
is an analytics system for the Open edX platform, bringing actionable data
about course and learner performance to instructors and site operators. It is primarily
a Tutor plugin that ties together data from the Open edX learning management system
and Studio using open source tools to aggregate and transform learning traces into data
visualizations.

See the `Aspects configuration documentation <https://docs.openedx.org/projects/openedx-aspects/en/latest/how-tos/production_configuration.html>`_

> to learn about setting up Aspects for your production environment.

Reviewer: ```suggestion
to learn about setting up Aspects for your production environment.

```

Deprecations & Removals
***********************

-  Badges app has been deprecated and removed from ``edx-platform``.
   See `[DEPR]: lms/djangoapps/badges <https://github.com/openedx/edx-platform/issues/31541>`_ .
-  - In edxapp, the Waffle switch ``ip.legacy`` is removed. See `[DEPR]: legacy_ip code and Waffle switch <https://github.com/openedx/edx-platform/issues/33733>`_ .

   -  Any deployment that has been relying on this legacy IP address
      option will need to switch to setting
      ``CLOSEST_CLIENT_IP_FROM_HEADERS`` appropriately. See `Nutmeg Announcement <https://openedx.atlassian.net/wiki/spaces/COMM/pages/3205201949/Nutmeg#CLOSEST_CLIENT_IP_FROM_HEADERS>`_ for
      details.
-  Asset Processing (webpack, collectstatic, etc.) using Paver Commands in edx-platform is now Deprecated and will not be available in Sumac

   - `[DEPR]: Asset processing in Paver`_.
   -  *Non deployment paver commands will be removed by Sumac*

-  The ``django-splash`` capability was removed from ``edx-platform`` and the relevant code has been archived.

   - https://github.com/openedx/public-engineering/issues/224

> Flags and toggles removed in Redwood

Reviewer: ```suggestion

Flags and toggles removed in Redwood
```

* ``BADGR_ENABLE_NOTIFICATIONS``, ``SEND_CERTIFICATE_REVOKED_SIGNAL``,
* ``blockstore.use_blockstore_app_api``,
* ``contentstore.enable_copy_paste_units``,
* ``course_apps.proctoring_settings_modal_view``,
* ``course_live.enable_big_blue_button``,
* ``course_live.enable_course_live``,
* ``courseware.learning_assistant``,
* ``discussions.enable_learners_stats``,
* ``discussions.enable_learners_tab_in_discussions_mfe``,
* ``discussions.enable_moderation_reason_codes``,
* ``discussions.enable_reported_content_email_notifications``,
* ``learner_recommendations.enable_course_about_page_recommendations``,
* ``learner_recommendations.enable_dashboard_recommendations``,
* ``student.enable_2u_recommendations``,
* ``student.enable_amplitude_recommendations``,
* ``student.enable_fallback_recommendations``,
* ``blockstore.use_blockstore_app_api`` are also not configurable anymore.

Developer Experience
********************

-  Asset Processing (webpack, collectstatic, etc.) using Paver Commands in edx-platform is now Deprecated and will not be available in Sumac

   -    - `[DEPR]: Asset processing in Paver`_.
   -  Non deployment paver commands will be removed by Sumac

> .. _ [DEPR]: Asset processing in Paver: https://github.com/openedx/edx-platform/issues/31895

Reviewer: ```suggestion
.. _[DEPR]: Asset processing in Paver: https://github.com/openedx/edx-platform/issues/31895
```

      the new course search feature in general `in the discussion forums <https://discuss.openedx.org/t/feedback-thread-new-course-search/13076>`_ or in the `#wg-product-core <https://openedx.slack.com/archives/C057J2D1WU9>`_ Slack channel.

Deprecations & Removals
***********************

-  Badges app has been deprecated and removed from ``edx-platform``.
   See `[DEPR]: lms/djangoapps/badges <https://github.com/openedx/edx-platform/issues/31541>`_ .
-  - In edxapp, the Waffle switch ``ip.legacy`` is removed. See `[DEPR]: legacy_ip code and Waffle switch <https://github.com/openedx/edx-platform/issues/33733>`_ .

   -  Any deployment that has been relying on this legacy IP address
      option will need to switch to setting
      ``CLOSEST_CLIENT_IP_FROM_HEADERS`` appropriately. See `Nutmeg Announcement <https://openedx.atlassian.net/wiki/spaces/COMM/pages/3205201949/Nutmeg#CLOSEST_CLIENT_IP_FROM_HEADERS>`_ for
      details.
-  Asset Processing (webpack, collectstatic, etc.) using Paver Commands in edx-platform is now Deprecated and will not be available in Sumac

>    - `[DEPR]: Asset processing in Paver`_.

Reviewer: ```suggestion
   - `[DEPR] Asset processing in Paver`_.
```

* ``BADGR_USERNAME``, ``BADGR_PASSWORD``,
* ``BADGR_TOKENS_CACHE_KEY``, ``BADGR_TIMEOUT``,
* ``BADGR_ENABLE_NOTIFICATIONS``, ``SEND_CERTIFICATE_REVOKED_SIGNAL``,
* ``blockstore.use_blockstore_app_api``,
* ``contentstore.enable_copy_paste_units``,
* ``course_apps.proctoring_settings_modal_view``,
* ``course_live.enable_big_blue_button``,
* ``course_live.enable_course_live``,
* ``courseware.learning_assistant``,
* ``discussions.enable_learners_stats``,
* ``discussions.enable_learners_tab_in_discussions_mfe``,
* ``discussions.enable_moderation_reason_codes``,
* ``discussions.enable_reported_content_email_notifications``,
* ``learner_recommendations.enable_course_about_page_recommendations``,
* ``learner_recommendations.enable_dashboard_recommendations``,
* ``student.enable_2u_recommendations``,
* ``student.enable_amplitude_recommendations``,
* ``student.enable_fallback_recommendations``,
* ``blockstore.use_blockstore_app_api`` are also not configurable anymore.

Developer Experience
********************

-  Asset Processing (webpack, collectstatic, etc.) using Paver Commands in edx-platform is now Deprecated and will not be available in Sumac

>    -    - `[DEPR]: Asset processing in Paver`_.

Reviewer: ```suggestion
   -    - `[DEPR] Asset processing in Paver`_.
```

   -  Non deployment paver commands will be removed by Sumac

> .. _[DEPR]: Asset processing in Paver: https://github.com/openedx/edx-platform/issues/31895

Reviewer: ```suggestion
.. _[DEPR] Asset processing in Paver: https://github.com/openedx/edx-platform/issues/31895
```

> Breaking Changes

Reviewer 2: Should we include this known issue in this section? https://github.com/openedx/wg-build-test-release/issues/380. 

Basically, this flag worked for the legacy Studio Home, improving performance by adding an organization search bar. But as the report mentions, it's not working correctly in the Studio Home MFE. The good thing is that this flag is not needed in the Studio Home MFE since we already have a search bar and pagination, which improves performance for those home pages with lots of courses. I wouldn't consider it a "breaking change" but a known issue for those who want to turn on the Studio Home MFE and are using the flag. 

Either way, I reported [this issue](https://github.com/openedx/frontend-app-course-authoring/issues/1110) to the Course Authoring maintainers so they're aware.

Reviewer 2: This issue is in this board too: https://github.com/orgs/openedx/projects/28/views/16. I don't know if we should add each issue or link to the board.

Author: Again I have no issue with this. I'd favor linking the board unless someone like you can tell me what to include and what not.

I'm just an overpaid secretary, I don't really understand much of what I'm working with here except in the most general (e.g. I know Python and Django) terms.

Author: OK.

I'm going to create a Known Issues section with the link to that board you referenced.

I'll also include this in the Breaking section.

****************

-  Deployers must ensure that their
   ``JWT_AUTH['JWT_PRIVATE_SIGNING_JWK']`` Django setting in LMS
   contains the full complement of private key numbers.*

   -  **Background:**\ *In LMS, we are switching from the pyjwkest
      library to PyJWT for signing JWTs. (pyjwkest is now unmaintained.)
      However, PyJWT has stricter requirements for the private key in
      ``JWT_PRIVATE_SIGNING_JWK``. Before you upgrade to Redwood, you
      will need to update this key using a script. Otherwise, JWT

>       signing will fail, and users will be unable to log in.*

Reviewer 3: I don't think this section needs to be in italics. Maybe the rendering will work better without it. 

Also, I think this would read more clearly if it were in the past tense. e.g. "In LMS, we switched from the pyjwkest library..."

Author: Good catch. I used a Confluence -> RTF and then RTF to RST converter on this and I suspect that's an artifact.

   -  **Steps:**

      1. Locate ``JWT_PRIVATE_SIGNING_JWK`` in your deployment

>          configuration.*

Reviewer 3: Each of the bullets in this section ends with an asterisk. Does that have some meaning?

Author: Ditto conversion artifact.

      2. Check if the JSON contains all of the following params: ``p``,
         ``q``, ``dp``, ``dq``, and ``qi``. If it does, you don’t need
         to do anything further. Otherwise, continue.*

      3. In your edx-platform virtualenv, run
         ``python3 scripts/jwk-precompute-params.py`` and follow the
         prompts. (It will ask you to paste in the current JSON.)*

      4. Update your config’s ``JWT_AUTH['JWT_PRIVATE_SIGNING_JWK']``
         with the output of the script.*

      5. You may wish to compare the contents of the private key before
         and after running the script, and verify that the only changes
         it has made to the contents of the JSON are that the ``p``,
         ``q``, ``dp``, ``dq``, and ``qi`` params have been added. (Some
         or all may already have been present.)*

   -  **Notes:**

      1. This should be done while you are still running Quince—it is
         safe to do ahead of the upgrade, and should not have any
         visible effect at that time.*

      2. This key must be handled very carefully. Anyone in possession
         of the key may impersonate users.*

> Learner Experiences

Author: @sarina Should we remove all these empty subheads that are covered in your doc? Or instead add links from here back to yours in the relevant spots?

         to do anything further. Otherwise, continue.

      3. In your edx-platform virtualenv, run
         ``python3 scripts/jwk-precompute-params.py`` and follow the
         prompts. (It will ask you to paste in the current JSON.)

      4. Update your config’s ``JWT_AUTH['JWT_PRIVATE_SIGNING_JWK']``
         with the output of the script.

      5. You may wish to compare the contents of the private key before
         and after running the script, and verify that the only changes
         it has made to the contents of the JSON are that the ``p``,
         ``q``, ``dp``, ``dq``, and ``qi`` params have been added. (Some
         or all may already have been present.)

   -  Notes:

      1. This should be done while you are still running Quince—it is
         safe to do ahead of the upgrade, and should not have any
         visible effect at that time.

      2. This key must be handled very carefully. Anyone in possession
         of the key may impersonate users.

- `studio_home.enable_global_staff_optimization flag no longer works in Studio MFE <https://github.com/openedx/wg-build-test-release/issues/380>`_

>    - Background: Basically, this flag worked for the legacy Studio Home, improving performance by

Author: > Does this flag still work if you don't update to the Studio MFE?

Excellent question, thanks for it! Asked on the issue and asked them to comment here if there are implications for the release notes, so I'm resolving this.

>    - Background: Basically, this flag worked for the legacy Studio Home, improving performance by

Reviewer: Please don't resolve conversations that aren't actually resolved. It has an impact on my review. We can resolve this conversation when we hear back from that ticket AND agree on a rewording.

>    - Background: Basically, this flag worked for the legacy Studio Home, improving performance by

Reviewer 2: It works with Legacy Studio; I tested it as part of the triage.

>    - Background: Basically, this flag worked for the legacy Studio Home, improving performance by

Author: I apologize for having resolve that and thank you for un-resolving.

>    - Background: Basically, this flag worked for the legacy Studio Home, improving performance by

Author: Given @mariajgrimaldi's answer does this need any changes please or are we good on this one?

>    - Background: Basically, this flag worked for the legacy Studio Home, improving performance by

Reviewer: I had made a suggestion on this line... did it disappear? Hmm

>    - Background: Basically, this flag worked for the legacy Studio Home, improving performance by

Reviewer: You incorporated it already :)

         to do anything further. Otherwise, continue.

      3. In your edx-platform virtualenv, run
         ``python3 scripts/jwk-precompute-params.py`` and follow the
         prompts. (It will ask you to paste in the current JSON.)

      4. Update your config’s ``JWT_AUTH['JWT_PRIVATE_SIGNING_JWK']``
         with the output of the script.

      5. You may wish to compare the contents of the private key before
         and after running the script, and verify that the only changes
         it has made to the contents of the JSON are that the ``p``,
         ``q``, ``dp``, ``dq``, and ``qi`` params have been added. (Some
         or all may already have been present.)

   -  Notes:

      1. This should be done while you are still running Quince—it is
         safe to do ahead of the upgrade, and should not have any
         visible effect at that time.

      2. This key must be handled very carefully. Anyone in possession
         of the key may impersonate users.

- `studio_home.enable_global_staff_optimization flag no longer works in Studio MFE <https://github.com/openedx/wg-build-test-release/issues/380>`_

>    - Background: Basically, this flag worked for the legacy Studio Home, improving performance by

Reviewer: ```suggestion
   - Background: This flag works for the legacy Studio Home, improving performance by
      adding an organization search bar.  This flag does not work in the Studio Home MFE, as it
      is not needed within the MFE. The MFE already have a search bar and pagination, which
     improves performance for those home pages with lots of courses. 
```

and remove all below text.

         with the output of the script.

      5. You may wish to compare the contents of the private key before
         and after running the script, and verify that the only changes
         it has made to the contents of the JSON are that the ``p``,
         ``q``, ``dp``, ``dq``, and ``qi`` params have been added. (Some
         or all may already have been present.)

   -  Notes:

      1. This should be done while you are still running Quince—it is
         safe to do ahead of the upgrade, and should not have any
         visible effect at that time.

      2. This key must be handled very carefully. Anyone in possession
         of the key may impersonate users.

- `studio_home.enable_global_staff_optimization flag no longer works in Studio MFE <https://github.com/openedx/wg-build-test-release/issues/380>`_
   - Background: This flag works for the legacy Studio Home, improving performance by
     adding an organization search bar.  This flag does not work in the Studio Home MFE, as it
     is not needed within the MFE. The MFE already have a search bar and pagination, which
     improves performance for those home pages with lots of courses.

>      adding an organization search bar. But as the report mentions, it's not working

Reviewer: ```suggestion
```

>      correctly in the Studio Home MFE. The good thing is that this flag is not needed

Reviewer: ```suggestion
```

>      in the Studio Home MFE since we already have a search bar and pagination, which

Reviewer: ```suggestion
```

>      improves performance for those home pages with lots of courses. I wouldn't consider

Reviewer: ```suggestion
```

>      it a "breaking change" but a known issue for those who want to turn on the Studio

Reviewer: ```suggestion
```

>      Home MFE and are using the flag.

Reviewer: ```suggestion
```

> Learner & Instructor Experiences

Reviewer: TODO in this section:

We need to add additional notes about how to enable specific learner/instructor experiences.

* For example, I'd put the note about Meileisearch here, since it's related to enabling a specific experience.
* We need instructions on enabling in-course sidebar navigation. @crathbun428 can you provide info here?
* Make Sections available independently of the course outline - I'm waiting to hear more about this
* Connect Teams in a course to Content Groups - I'm waiting to hear more about this

Reviewer: @mariajgrimaldi - Santiago says you can help with this one `Make Sections available independently of the course outline - I'm waiting to hear more about this`

Reviewer 2: @sarina: I can help with `Connect Teams in a course to Content Groups`, but not with` Make Sections available independently of the course outline`, I'm not sure what that is.

I'll be on it in the next few hours.

Reviewer 2: Alright. So for `Connect Teams in a course to Content Groups`:
1. Enable teams feature by turning on `teams.enable_teams_app`
2. Then, turn on the  `teams.content_groups_for_teams` waffle flag for everyone or for specific courses with a waffle flag course overrides

And for `Make Sections available independently of the course outline` enable this feature flag:
`FEATURES["ENABLE_HIDE_FROM_TOC_UI"] = True` to enable the feature system-wide.

Reviewer: Thanks @mariajgrimaldi I added those as a suggestion below. Still need instructions about enabling the course sidebar (I know not your project, I'll get the info from Chelsea's team) but we're good on the other two!

Reviewer 2: In case it helps, I found this waffle flag: `courseware.enable_navigation_sidebar` linked in this PR https://github.com/openedx/frontend-app-learning/pull/1398

Reviewer: You're the best! I'll verify with Glib that this is all that needs to be done.

********************************

> See the Product Release Notes.

Reviewer: ```suggestion
For in-depth information on new learner and instructor facing features in the Redwood release, please see the Feature-Based Release Notes. Instructions on how to enable those features are as follows.

Connect Teams in a course to Content Groups
===================================

#. Go to your site's Django Admin Panel

#. Enable the teams feature by turning on the waffle flag: ``teams.enable_teams_app``

#. Then, turn on the ``teams.content_groups_for_teams`` waffle flag for everyone or specific courses with a waffle flag course overrides

Make Sections available independently of the course outline
=============================================

Enable this feature flag: ``FEATURES["ENABLE_HIDE_FROM_TOC_UI"] = True`` to your deployment configurations to enable the feature system-wide.
```

> ******************************

Reviewer: ```suggestion
```

> Instructor Experiences

Reviewer: ```suggestion
```

Reviewer: Also need to remove the underline one line below (I can't make a comment it's too far down in the diff)

Author: Did that in next revision. Thanks!

Administrators & Operators
**************************

> Settings and Toggles

Reviewer 2: I generated this list of settings/toggles/flags with an edx-platform docs diff and a really naive script:

### ➕ Added
1. ANONYMOUS_SURVEY_REPORT
2. EDX_DRF_EXTENSIONS[ENABLE_JWT_AND_LMS_USER_EMAIL_MATCH]
3. ENABLE_AUTO_GENERATED_USERNAME
4. ENFORCE_SESSION_EMAIL_MATCH
5. EVENT_BUS_PRODUCER_CONFIG['org.openedx.content_authoring.course.catalog_info.changed.v1']
6. EVENT_BUS_PRODUCER_CONFIG['org.openedx.learning.course.unenrollment.completed.v1']
7. EVENT_BUS_PRODUCER_CONFIG['org.openedx.learning.xblock.skill.verified.v1']
8. FEATURES['ENABLE_BLAKE2B_HASHING']
9. FEATURES['ENABLE_CERTIFICATES_INSTRUCTOR_MANAGE]
10. FEATURES['ENABLE_COURSEWARE_SEARCH_VERIFIED_REQUIRED']
11. FEATURES['ENABLE_GRADING_METHOD_IN_PROBLEMS']
12. FEATURES['ENABLE_HIDE_FROM_TOC_UI']
13. FEATURES['ENABLE_HOME_PAGE_COURSE_API_V2']
14. FEATURES['ENABLE_LTI_PII_ACKNOWLEDGEMENT']
15. FEATURES['ENABLE_ORA_SELECTABLE_LEARNER_WAITING_REVIEW']
16. FEATURES['ENABLE_SEND_XBLOCK_LIFECYCLE_EVENTS_OVER_BUS']
17. FEATURES['SEND_LEARNING_CERTIFICATE_LIFECYCLE_EVENTS_TO_BUS']
18. JWT_AUTH_ADD_KID_HEADER:
19. SEND_TRACKING_EVENT_EMITTED_SIGNAL
20. SURVEY_REPORT_ENABLE
21. agreements.enable_lti_pii_acknowledgement
22. commerce.transition_to_coordinator.checkout
23. commerce.transition_to_coordinator.refund
24. content_tagging.disabled
25. contentstore.new_studio_mfe.use_new_certificates_page
26. contentstore.new_studio_mfe.use_new_group_configurations_page
27. contentstore.new_studio_mfe.use_new_textbooks_page
28. course_home.new_discussion_sidebar_view
29. teams.content_groups_for_teams
30. courseware.always_open_auxiliary_sidebar
31. courseware.disable_navigation_sidebar_blocks_caching
32. courseware.discovery_default_language_filter
33. courseware.enable_navigation_sidebar
34. discussions.enable_reported_content_notifications
35. enterprise.enterprise_groups_v1
36. enterprise.feature_prequery_search_suggestions
37. notifications.enable_coursewide_notifications
38. notifications.enable_email_notifications
39. notifications.enable_notifications_filters
40. notifications.enable_ora_staff_notifications
41. student.redirect_to_courseware_after_enrollment
42. studio.enable_course_update_notifications
43. user_tours.tours_disabled
44. video_config.transcript_feedback
45. video_config.xpert_translations_ui

### ❌ Removed
1. BADGR_ENABLE_NOTIFICATIONS
2. FEATURES['ENABLE_OPENBADGES']
3. FEATURES['ENABLE_SEND_XBLOCK_EVENTS_OVER_BUS']
4. SEND_CERTIFICATE_CREATED_SIGNAL
5. SEND_CERTIFICATE_REVOKED_SIGNAL
6. blockstore.use_blockstore_app_api
7. contentstore.enable_copy_paste_units
8. course_apps.proctoring_settings_modal_view
9. course_live.enable_big_blue_button
10. course_live.enable_course_live
11. courseware.learning_assistant
12. discussions.enable_learners_stats
13. discussions.enable_learners_tab_in_discussions_mfe
14. discussions.enable_moderation_reason_codes
15. discussions.enable_reported_content_email_notifications
16. learner_recommendations.enable_course_about_page_recommendations
17. learner_recommendations.enable_dashboard_recommendations
18. student.enable_2u_recommendations
19. student.enable_amplitude_recommendations
20. student.enable_fallback_recommendations

That's what I wanted to automate [in this issue](https://github.com/openedx/wg-build-test-release/issues/344), but I didn't get to it in time. So I'm adding this here just for reference, I'm not sure how accurate this list is.

Author: OOH thank you! Could you please link your script, *no matter how naive* to that issue? I'll work on this. Honestly this toggle business is my biggest pain point in this process and I find it a bit stressful currently.

Reviewer 2: This sounds unbelievable, but my hard drive died a couple of weeks ago, and I lost all the work I had on there 😭

Reviewer 2: Let me see what I can find. I'll ping you :)

Author: These have been integrated into the next revision. Thanks!

Reviewer: Hi @feoh I see the Added flags in the doc, but I don't see all of the Removed flags. Some, but not all, appear to be mentioned in the DEPR section. Could you please review the Removed flags that María listed above, and ensure they all appear in the release notes? Thank you 🙇🏻‍♀️ 💯

Author: This is done in the current revision. Also noticed that my markup was all over the place in that section and should be fixed now. Let's hope ReadTheDocs agrees :)

Author: Wow that made - a mess. Redoing. I remain allergic to ReStructuredText apparently :(

Author: OK that's all fixed and all the removed toggles are there looking reasonable.

Reviewer: Thanks, looks great :)

====================

> Waffle flags:

Reviewer 3: are these all new flags?

Author: It's a fine question. I'm going to throw out the current list and use the one @mariajgrimaldi generated. I don't trust my work. I was trying to guess and that's NEVER what you want.

`contentstore.new_studio_mfe.use_new_textbooks_page <https://github.com/openedx/edx-platform/blob/f256684646aec6fd0d5519c6900ec99077e7db50/cms/djangoapps/contentstore/toggles.py#L504>`_
`contentstore.new_studio_mfe.use_new_unit_page (disabled) <https://github.com/openedx/edx-platform/blob/f256684646aec6fd0d5519c6900ec99077e7db50/cms/djangoapps/contentstore/toggles.py#L444>`_
`contentstore.new_studio_mfe.use_new_updates_page <https://github.com/openedx/edx-platform/blob/f256684646aec6fd0d5519c6900ec99077e7db50/cms/djangoapps/contentstore/toggles.py#L324>`_
`contentstore.new_studio_mfe.use_new_video_uploads_page (disabled) <https://github.com/openedx/edx-platform/blob/f256684646aec6fd0d5519c6900ec99077e7db50/cms/djangoapps/contentstore/toggles.py#L404>`_
`new_studio_mfe.use_new_home_page <https://github.com/openedx/edx-platform/blob/f256684646aec6fd0d5519c6900ec99077e7db50/cms/djangoapps/contentstore/toggles.py#L225>`_

Course authoring MFE config:

ENABLE_UNIT_PAGE: "false"
ENABLE_NEW_EDITOR_PAGES: "true"
ENABLE_TAGGING_TAXONOMY_PAGES: "true"
ENABLE_ASSETS_PAGE: "true"

> ENABLE_HOME_PAGE_COURSE_API_V2: "true"

Reviewer 3: Can these be linked to their context, like the other waffles?

Author: I'm throwing out my current list in favor of what Maria posted.

Other Operator Changes
======================

-  The default minimum password length has been updated from 2
   characters to 8 characters. `(PR) <https://github.com/openedx/edx-platform/pull/33373>`_.

>    -  If you have an existing password, this change along will not

Reviewer 3: ```suggestion
   -  If you have an existing password, this change alone will not
```

      force you to update it. However if you reset your password or go
      to change it, you'll have to conform to the new guidelines. If you
      would like to force people to update their password, you'll

>       probably want to take a look at .. _the password_policy plugin and its settings: https://github.com/openedx/edx-platform/blob/2033dcf6ace133719aaeb72dc5dd6ee521a7ac42/openedx/core/djangoapps/password_policy/settings/common.py#L13 .*

Reviewer 3: this isn't formatted like the other links. Should it be something like this?

```suggestion
      probably want to take a look at `the password_policy plugin and its settings <https://github.com/openedx/edx-platform/blob/2033dcf6ace133719aaeb72dc5dd6ee521a7ac42/openedx/core/djangoapps/password_policy/settings/common.py#L13>`_
```

-  The following `requirements update <https://github.com/openedx/credentials/commit/1cd7c25c04a955aa9aaa263fb40ebd3f73d0937e>`_ into credentials might have implications for anyone
   who has a massive ``usersocialauth`` table.  This is because that
   table grows endlessly, and the migrations on the table caused by
   updating the ``social-auth-app-django`` package can run out of
   memory. If maintainers have migration failures on this upgrade, they
   should run the management command `truncate_social_auth <https://github.com/openedx/credentials/blob/master/credentials/apps/core/management/commands/truncate_social_auth.py>`_.*

   -  This will remove all entries from the ``usersocialauth`` table
      that haven't been updated in 90 days, which makes the size of the
      table tractable for the dependency's migration. This is harmless
      in the ``Credentials`` IDA.*

-  The scripts related to user retirement across all services
   have been moved to the ``edx-platform`` repository. If you’ve been

>    using the \`tubular repo to run retirement scripts you should update

Reviewer 3: ```suggestion
   using the tubular repo to run retirement scripts you should update
```

      ``BADGR_USERNAME``, ``BADGR_PASSWORD``,
      ``BADGR_TOKENS_CACHE_KEY``, ``BADGR_TIMEOUT``,
      ``BADGR_ENABLE_NOTIFICATIONS`` are also not configurable anymore.*

-  * In edxapp, the Waffle switch ``ip.legacy`` is removed. See `[DEPR]: legacy_ip code and Waffle switch <https://github.com/openedx/edx-platform/issues/33733>`_ .*

   -  *Any deployment that has been relying on this legacy IP address
      option will need to switch to setting
      ``CLOSEST_CLIENT_IP_FROM_HEADERS`` appropriately. See `Nutmeg Announcement <https://openedx.atlassian.net/wiki/spaces/COMM/pages/3205201949>`_ for
      details.*

-  *Asset Processing (webpack, collectstatic, etc.) using Paver Commands in edx-platform is now Deprecated and will not be available in Sumac*
   - `[DEPR]: Asset processing in Paver <https://github.com/openedx/edx-platform/issues/31895>`_.
   -  *This should cover everything that a site operator needs to do as a part of deployment.*
   -  *Non deployment paver commands will be removed by Sumac*

-  *The django-splash capability was removed from edx-platform and the
   relevant code has been archived.*

>    - https://github.com/openedx/public-engineering/issues/224

Reviewer 2: Do we have a section for "known issues" where we can include some of the issues found during testing that we didn't get to close before the release?

Author: We don't but we should! I'll add it with my next revision.

Here's the thing though: Please tell me it'll be easy for me to get links to those, too much of the rest of the information gathering for this is like tooth extraction :)

Reviewer 2: I'm not sure what folks think but I believe using this list is enough: https://github.com/orgs/openedx/projects/28/views/16

Reviewer 2: Those are known issues from "release testing", they're not filtered by redwood or any other release, so we should add a clarification note if it's gonna be used as is.

Administrators & Operators
**************************

Settings and Toggles
====================

Waffle flags Added In Redwood:

> `ANONYMOUS_SURVEY_REPORT <https://github.com/openedx/edx-platform/blob/7d11c889bbbf55dfa69c734122de72d83c1893bf/lms/envs/common.py#L5523>`_

Author: They're now bulleted. I agree it reads much better. Thanks.

`enterprise.enterprise_groups_v1 <https://github.com/openedx/edx-enterprise/blob/007abaf5b10707607d47a9f9d89572b36d18b8e2/enterprise/toggles.py#L34>`_
`enterprise.feature_prequery_search_suggestions <https://github.com/openedx/edx-enterprise/blob/007abaf5b10707607d47a9f9d89572b36d18b8e2/enterprise/toggles.py#L22>`_
`EVENT_BUS_PRODUCER_CONFIG['org.openedx.content_authoring.course.catalog_info.changed.v1'] <https://github.com/openedx/edx-platform/blob/7d11c889bbbf55dfa69c734122de72d83c1893bf/cms/envs/common.py#L2849>`_
`EVENT_BUS_PRODUCER_CONFIG['org.openedx.learning.course.unenrollment.completed.v1'] <https://github.com/openedx/edx-platform/blob/7d11c889bbbf55dfa69c734122de72d83c1893bf/lms/envs/common.py#L5428>`_
`EVENT_BUS_PRODUCER_CONFIG['org.openedx.learning.xblock.skill.verified.v1'] <https://github.com/openedx/edx-platform/blob/7d11c889bbbf55dfa69c734122de72d83c1893bf/lms/envs/common.py#L5443>`_

Other Operator Changes
======================

-  The default minimum password length has been updated from 2
   characters to 8 characters. `(PR) <https://github.com/openedx/edx-platform/pull/33373>`_.

>    -  If you have an existing password, this change alone will not

Reviewer: @feoh - are you seeing these comments? Are you using the files view (https://github.com/openedx/docs.openedx.org/pull/507/files)? I see quite a few suggestion comments like this still waiting to be accepted or discussed.

   -  *``accomplishments_shared`` field is removed from payloads and
      settings*

   -  *``ENABLE_OPENBADGES`` is no longer available for configuration in
      ``FEATURES``*

   -  *``BADGING_BACKEND``, ``BADGR_BASE_URL``, ``BADGR_ISSUER_SLUG``,
      ``BADGR_USERNAME``, ``BADGR_PASSWORD``,
      ``BADGR_TOKENS_CACHE_KEY``, ``BADGR_TIMEOUT``,
      ``BADGR_ENABLE_NOTIFICATIONS`` are also not configurable anymore.*

-  * In edxapp, the Waffle switch ``ip.legacy`` is removed. See `[DEPR]: legacy_ip code and Waffle switch <https://github.com/openedx/edx-platform/issues/33733>`_ .*

   -  *Any deployment that has been relying on this legacy IP address
      option will need to switch to setting
      ``CLOSEST_CLIENT_IP_FROM_HEADERS`` appropriately. See `Nutmeg Announcement <https://openedx.atlassian.net/wiki/spaces/COMM/pages/3205201949>`_ for
      details.*

-  *Asset Processing (webpack, collectstatic, etc.) using Paver Commands in edx-platform is now Deprecated and will not be available in Sumac*
   - `[DEPR]: Asset processing in Paver <https://github.com/openedx/edx-platform/issues/31895>`_.

>    -  *This should cover everything that a site operator needs to do as a part of deployment.*

Reviewer: I don't understand what information this line is imparting

Author: Great question! I'll delete it. Let me know if that was the wrong call please.

   -  *Non deployment paver commands will be removed by Sumac*

-  *The django-splash capability was removed from edx-platform and the
   relevant code has been archived.*
   - https://github.com/openedx/public-engineering/issues/224

> Developer Experience

Reviewer: For dev exp I would re-mention the note about asset processing in Paver. I'll check on my end to see if anything else needs to be added.

Author: OK. I repeated the paver entry from breaking in developer experience. Please let me know if you had something else in mind, being as specific as possible please because I'm still an OpenEdX noob :)

Author: Also for some reason ReST won't let me duplidate that bullet as-is, so had to remove the issue link?

Reviewer: I think I know how to work around that, but I can address post-merge.

> 

Reviewer: for Researcher & Data, we should add a link to the product notes about the Aspects release, and detail what it takes to install it on your instance.

Author: OK please forgive my ignorance I have no idea what that means :) Who should I talk to to get that excellent suggestion transmuted into text I can include?

Reviewer: @bmtcril  what should the developer & operator release notes include about Aspects (eg how to install/enable/administer)

Author: FWIW I *THINK* this is the only outstanding change we need to make so we can be ready for tomorrow's release. Thanks for the feedback we've gotten thus far all!
